test/distributed/_tensor/test_dtensor.py

```diff
@@ -825,6 +825,89 @@ def test_split_tensor_1D(self) -> None:
                 ]
                 assert_array_equal(expected_is_tensor_empty, is_tensor_empty)
 
+    def _init_mesh_tensor(self):
+        tensor = torch.tensor([1.0 * i for i in range(16)], device=self.device_type)
+        mesh = init_device_mesh(
+            self.device_type,
+            mesh_shape=(2, self.world_size // 2),
+            mesh_dim_names=("p1", "p2"),
+        )
+        return mesh, tensor
+
+    def _check_dtensor(
+        self, dtensor, mesh, placements, expected_size, expected_start_value
+    ):
+        local_tensor = dtensor._local_tensor
+        self.assertEqual(local_tensor.size(), expected_size)
+        self.assertEqual(mesh, dtensor.device_mesh)
+        self.assertEqual(placements, dtensor.placements)
+        self.assertEqual(
+            local_tensor,
+            torch.tensor(
+                [1.0 * (expected_start_value + i) for i in range(expected_size[0])],
+                device=self.device_type,
+            ),
+        )
+
+    @with_comms
+    def test_distribute_dtensor_fsdp_tp(self) -> None:
+        mesh, tensor = self._init_mesh_tensor()
+
+        # Mimic FSDP + TP, p1 == FSDP, p2 == TP
+        dtensor_tp = distribute_tensor(tensor, mesh["p2"], placements=[Shard(0)])
+        dtensor_dp_tp = distribute_tensor(
+            dtensor_tp,
+            mesh["p1"],
+            placements=[Shard(0)],
```

Comment:
hmmm I think this should error out? We can only further distribute a DTensor when the mesh of the DTensor is in the left of the parent mesh, not the right.

For FSDP + TP case, we would need to enable the `strided sharding` in the `Shard` placement, @XilunWu is working on that currently